src/System.Diagnostics.TraceSource/tests/TraceSourceClassTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.Diagnostics.TraceSourceTests
+{
+    using Method = TestTraceListener.Method;
+
+    public class TraceSourceTests
+    {
+        [Fact]
+        public void DefaultListener()
+        {
+            var trace = new TraceSource("TestTraceSource");
+            Assert.True(trace.Listeners.Count == 1);
+            Assert.IsType<DefaultTraceListener>(trace.Listeners[0]);
+        }
+
+        [Fact]
+        public void DefaultLevel()
+        {
+            var trace = new TraceSource("TestTraceSource");
+            Assert.True(trace.Switch.Level == SourceLevels.Off);
+        }
+
+        [Theory]
+        [InlineData(SourceLevels.Off, TraceEventType.Critical, 0)]
+        [InlineData(SourceLevels.Critical, TraceEventType.Critical, 1)]
+        [InlineData(SourceLevels.Critical, TraceEventType.Error, 0)]
+        [InlineData(SourceLevels.Error, TraceEventType.Error, 1)]
+        [InlineData(SourceLevels.Error, TraceEventType.Warning, 0)]
+        [InlineData(SourceLevels.Warning, TraceEventType.Warning, 1)]
+        [InlineData(SourceLevels.Warning, TraceEventType.Information, 0)]
+        [InlineData(SourceLevels.Information, TraceEventType.Information, 1)]
+        [InlineData(SourceLevels.Information, TraceEventType.Verbose, 0)]
+        [InlineData(SourceLevels.Verbose, TraceEventType.Verbose, 1)]
+        [InlineData(SourceLevels.All, TraceEventType.Critical, 1)]
+        [InlineData(SourceLevels.All, TraceEventType.Verbose, 1)]
+        // NOTE: tests to cover a TraceEventType value that is not in CoreFX (0x20 == TraceEventType.Start in 4.5)
+        [InlineData(SourceLevels.Verbose, (TraceEventType)0x20, 0)]
+        [InlineData(SourceLevels.All, (TraceEventType)0x20, 1)]
+        public void SwitchLevel(SourceLevels sourceLevel, TraceEventType messageLevel, int expected)
+        {
+            var trace = new TraceSource("TestTraceSource");
+            var listener = new TestTraceListener();
+            trace.Listeners.Add(listener);
+            trace.Switch.Level = sourceLevel;
+            trace.TraceEvent(messageLevel, 0);
+            Assert.Equal(expected, listener.GetCallCount(Method.TraceEvent));
+        }
+
+        [Fact]
+        public void TraceEvent()
+        {
+            var trace = new TraceSource("TestTraceSource", SourceLevels.All);
+            var listener = new TestTraceListener();
+            trace.Listeners.Add(listener);
+            trace.TraceEvent(TraceEventType.Verbose, 0);
```

Comment:
One of the TraceEvent methods you modified in TraceSource.cs isn't getting exercised.  Could you add another call here that'll TraceEvent(TraceEvent, 0, "message") so that it gets hit?